Number_types/include/CGAL/Quotient.h

```diff
@@ -685,12 +785,239 @@ template < class NT > class Real_embeddable_traits_quotient_base< Quotient<NT> >
 
     class To_interval
       : public CGAL::cpp98::unary_function< Type, std::pair< double, double > > {
+
       public:
+
+        bool are_correct_bounds( const double l, const double u, const Type& x ) const {
+
+          if (
+            CGAL::abs(l) == std::numeric_limits<double>::infinity() ||
+            CGAL::abs(u) == std::numeric_limits<double>::infinity() ||
+            CGAL::abs(l) == 0.0 ||
+            CGAL::abs(u) == 0.0) {
```

Comment:
Those conditions are strange...

Reply:
The condition with `inf` comes from the fact that when we convert `l` and `u` to `Type`, which is is `Quotient<cpp_int>`, and `l` or `u` are `inf`, this `Type` construction fails because `cpp_int` does not accept the `inf`. The condition with 0 is similar. In some cases, when `l` is 0, `u` is a small value like `min_denorm` and again `Type` construction fails on this value, so I skip all cases with 0 to avoid this.

Reply:
I see, so l=-inf or u=+inf is ok and we don't want to convert in that case. I am not convinced for 0 though. We should be able to convert from denorm_min, or if we cannot then we may need to add some special code to handle that case.

Reply:
For `denorm_min` it fails [here](https://github.com/danston/cgal/blob/2e976c4048101036cf59c0235dde36b5c129bef8/Number_types/include/CGAL/double.h#L206).

Reply:
I opened #5982 about that.